browser/ui/webui/settings/brave_extensions_manifest_v2_browsertest.cc

```diff
@@ -173,3 +180,63 @@ IN_PROC_BROWSER_TEST_F(BraveExtensionsManifestV2BrowserTest,
   EXPECT_FALSE(IsExtensionInstalled());
   EXPECT_FALSE(IsExtensionEnabled());
 }
+
+class BraveExtensionsManifestV2InstallerBrowserTest
```

Comment:
we should also have a unit test for the installer code

Reply:
browser tests are slow, we should unit test edge cases including extensions that are not properly signed and only use browser tests for the high level functionality

Reply:
We inherit such tests from the chromium codebase. If an extension has invalid content or an invalid signature it can't be installed and this behaviour is tested elsewhere.

Reply:
Some things need browser tests, but other things can be adequately tested without them. You should try to keep browser tests to a minimum so they only cover the parts that are mocked in unit tests. So for instance if you have a browser test for the full installation, you can unit test edge cases like installing different Brave extensions, ensuring that non-brave extensions still install correctly with the upstream key. Other cases like cancel and error can probably be unit tested since the only difference is the response from the installer.